content/en/integrations/guide/source-code-integration.md

```diff
@@ -376,125 +460,20 @@ If your build process is executed in CI within a Docker container, perform the f
-180 tracked file paths will be reported.
-Successfully uploaded tracked files in 1.358 seconds.
-Syncing GitDB...
-Successfully synced git DB in 3.579 seconds.
-✅ Uploaded in 5.207 seconds.
-```
-
-[1]: https://gitlab.com
-[2]: https://github.com/DataDog/datadog-ci/tree/master/src/commands/git-metadata
-[3]: /logs/error_tracking/backend/?tab=serilog#setup
-[4]: /integrations/guide/source-code-integration/?tab=continuousprofiler#links-to-git-providers
-[5]: /serverless/aws_lambda/configuration/?tab=datadogcli#link-errors-to-your-source-code
-[6]: /tests/developer_workflows/#open-tests-in-github-and-your-ide
-[7]: /security/application_security/
-
-{{% /tab %}}
-{{% tab "Other Git Providers" %}}
-
-<div class="alert alert-warning">
-Repositories on self-hosted instances or private URLs are not supported out-of-the-box by the source code integration. To enable this feature, <a href="/help">contact Support</a>.
-</div>
-
-To link telemetry with your source code, upload your repository metadata with the [`datadog-ci git-metadata upload`][1] command. `datadog-ci v2.10.0` or later is required.
-
-When you run `datadog-ci git-metadata upload` within a Git repository, Datadog receives the repository URL, the commit SHA of the current branch, and a list of tracked file paths.
-
-Run this command for every commit that you need to be synchronized with Datadog.
-
-### Validation
-
-To ensure the data is being collected, run `datadog-ci git-metadata upload` in your CI pipeline.
-
-You can expect to see the following output:
-
-```
-Reporting commit 007f7f466e035b052415134600ea899693e7bb34 from repository git@my-git-server.com:my-org/my-repository.git.
-180 tracked file paths will be reported.
-Successfully uploaded tracked files in 1.358 seconds.
-Syncing GitDB...
-Successfully synced git DB in 3.579 seconds.
-✅ Uploaded in 5.207 seconds.
-```
-
-[1]: https://github.com/DataDog/datadog-ci/tree/master/src/commands/git-metadata
-{{% /tab %}}
-{{< /tabs >}}
-
 ## Usage
 
-### Links to Git providers
+### Links to Git providers & code snippets
 
 {{< tabs >}}
 {{% tab "Error Tracking" %}}
 You can see links from stack frames to their source repository in [Error Tracking][1].
 
 1. Navigate to [**APM** > **Error Tracking**][2].
 2. Click on an issue. The **Issue Details** panel appears on the right.
-3. Under **Latest Event**, click the **View** button on the right of a frame or select **View file**, **View Git blame**, or **View commit** to be redirected to your source code management tool.
+3. Under **Latest Event**, if you're using the GitHub or GitLab integrations, click **Connect to preview** on stack frames. You can see inline code snippets directly in the stack trace. Otherwise, you can always click the **View** button on the right of a frame or select **View file**, **View Git blame**, or **View commit** to be redirected to your source code management tool.
```

Comment:
```suggestion
3. Under **Latest Event**, if you're using the GitHub or GitLab integrations, click **Connect to preview** on stack frames. You can see inline code snippets directly in the stack trace. Otherwise, you can click the **View** button on the right of a frame or select **View file**, **View Git blame**, or **View commit** to be redirected to your source code management tool.
```